### PR TITLE
fix(SF-1132): Make `sort` property optional when parsing search state

### PR DESCRIPTION
**Pull request by jrmykolyn**
Update 'searchStateToRequest' to check for presence/truthy-ness of 'urlSort' before using as arg.

### CODE REVIEW

File: test/unit/services/urlUtils.ts [hunk 1]
```diff
@@ -130,23 +130,25 @@ suite('URL Service', ({ expect, spy, stub }) => {
     });
 
     describe('searchStateToRequest()', () => {
+      let collection = 'a collection';
+      let page = 324;
+      let pageSize = 25;
+      let query = 'dress';
+      let refinements = [1,2,3,4,5];
+      let skip = 30;
+      let sort = { a: 'b' };
```
Comment (jmbtutor) [inline test/unit/services/urlUtils.ts]: You should declare these here and then assign them in a `beforeEach` to make sure that your tests are isolated. If they don't change, you can make them `const` instead.

File: src/services/urlUtils.ts [hunk 2]
```diff
@@ -68,22 +68,18 @@ namespace UrlUtils {
       refinements: urlRefinements = [],
       sort: urlSort = Selectors.sort(store),
     } = state;
-    let request: Partial<Request> = {};
-    const pageSize = Adapters.Request.clampPageSize(page, urlPageSize);
-    const skip = Adapters.Request.extractSkip(page, pageSize);
-    const refinements = <any>urlRefinements.map((refinement) =>
+    let request: Partial<Request> = { collection, query };
```
Comment (jmbtutor) [inline src/services/urlUtils.ts]: `request` does not get reassigned, so this can be `const`.

File: test/unit/services/urlUtils.ts (hunk 1 above)
Comment (jrmykolyn) [reply test/unit/services/urlUtils.ts]: Noted, will do.

File: src/services/urlUtils.ts (hunk 2 above)
Comment (jrmykolyn) [reply src/services/urlUtils.ts]: Confirmed, will do.

File: src/services/urlUtils.ts [hunk 3]
```diff
@@ -68,22 +68,18 @@ namespace UrlUtils {
       refinements: urlRefinements = [],
       sort: urlSort = Selectors.sort(store),
     } = state;
-    let request: Partial<Request> = {};
-    const pageSize = Adapters.Request.clampPageSize(page, urlPageSize);
-    const skip = Adapters.Request.extractSkip(page, pageSize);
-    const refinements = <any>urlRefinements.map((refinement) =>
+    const request: Partial<Request> = { collection, query };
+    request.pageSize = Adapters.Request.clampPageSize(page, urlPageSize);
```
[the remark this answers is not in the document]
Comment (jrmykolyn) [reply src/services/urlUtils.ts]: Will do 👍.

File: src/services/urlUtils.ts [hunk 4]
```diff
@@ -68,22 +68,27 @@ namespace UrlUtils {
...
       sort: urlSort = Selectors.sort(store),
     } = state;
-    let request: Partial<Request> = {};
-    const pageSize = Adapters.Request.clampPageSize(page, urlPageSize);
-    const skip = Adapters.Request.extractSkip(page, pageSize);
-    const refinements = <any>urlRefinements.map((refinement) =>
-      Adapters.Request.extractRefinement(refinement.field, <any>refinement)
-    );
-    const sort = <any>Adapters.Request.extractSort(urlSort);
+    const request: Partial<Request> = { collection, query };
```
Comment (Vic-Dev) [inline src/services/urlUtils.ts]: Should also check if these exist before adding them

File: src/services/urlUtils.ts [hunk 5]
```diff
@@ -96,22 +101,27 @@ namespace UrlUtils {
...
```
Comment (Vic-Dev) [inline src/services/urlUtils.ts]: Same thing here

File: test/unit/services/urlUtils.ts [hunk 6]
```diff
@@ -130,23 +130,35 @@ suite('URL Service', ({ expect, spy, stub }) => {
...
           page,
           pageSize,
           query,
           refinements,
-          sort
+          sort,
         };
         const store: any = { c: 'd' };
-        const skip = 30;
+
```
Comment (Vic-Dev) [inline test/unit/services/urlUtils.ts]: Remove newline

File: test/unit/services/urlUtils.ts [hunk 7]
```diff
@@ -162,19 +174,13 @@ suite('URL Service', ({ expect, spy, stub }) => {
...
       it('should return search request based off store', () => {
-        const collection = 'a collection';
-        const page = 324;
-        const pageSize = 25;
-        const query = 'dress';
-        const sort = { a: 'b' };
         const state: any = {};
         const store: any = { c: 'd' };
-        const skip = 30;
+
```
Comment (Vic-Dev) [inline test/unit/services/urlUtils.ts]: Remove newline

**Pull request merged**